src/main/java/com/denizenscript/depenizen/bukkit/commands/mythicmobs/MythicSkillCommand.java

```diff
@@ -3,113 +3,104 @@
     // You may not have both the Entity and Location targets, one or the other.
     // The MythicMob configuration must be configured to account for this.
+    // You may also specify an EntityTag as the trigger of the skill, an
+    // origin of the skill, a Map of skill parameters to be set for the skill, and a Map of skill variables to be set for the skill.
+    // The value of the variables can be obtained from MythicMobs through the placeholder <skill.var.x>, where x is the variable ID (the key of the map).
     //
     // @Usage
     // Used to make the player use the MythicMob skill frostbolt on their target.
-    // - mythicskill <player> frostbolt <player.target>
+    // - mythicskill frostbolt <player.target> casters:<player>
     //
     // -->
 
     @Override
-    public void parseArgs(ScriptEntry scriptEntry) throws InvalidArgumentsException {
-        for (Argument arg : scriptEntry) {
-            if (!scriptEntry.hasObject("casters")
-                    && arg.matchesPrefix("casters")
-                    && arg.matchesArgumentList(EntityTag.class)) {
-                scriptEntry.addObject("casters", arg.asType(ListTag.class).filter(EntityTag.class, scriptEntry));
-            }
-            else if (!scriptEntry.hasObject("skill")
-                    && MythicMobsBridge.skillExists(arg.asElement().asString())) {
-                scriptEntry.addObject("skill", arg.asElement());
-            }
-            else if (!scriptEntry.hasObject("entity_targets")
-                    && arg.matchesArgumentList(EntityTag.class)) {
-                scriptEntry.addObject("entity_targets", arg.asType(ListTag.class).filter(EntityTag.class, scriptEntry));
-            }
-            else if (!scriptEntry.hasObject("location_targets")
-                    && arg.matchesArgumentList(LocationTag.class)) {
-                scriptEntry.addObject("location_targets", arg.asType(ListTag.class).filter(LocationTag.class, scriptEntry));
-            }
-            else if (!scriptEntry.hasObject("power")
-                    && arg.matchesFloat()) {
-                scriptEntry.addObject("power", arg.asElement());
-            }
-            else {
-                arg.reportUnhandled();
-            }
-        }
-        if (!scriptEntry.hasObject("skill")) {
-            throw new InvalidArgumentsException("Must specify a valid skill.");
-        }
-        if (scriptEntry.hasObject("entity_targets") && scriptEntry.hasObject("location_targets")) {
-            throw new InvalidArgumentsException("Cannot have both entity and location targets.");
-        }
-        if (!scriptEntry.hasObject("casters")) {
-            scriptEntry.defaultObject("casters", Utilities.entryDefaultEntityList(scriptEntry, true));
-        }
-        scriptEntry.defaultObject("power", new ElementTag(0));
+    public void addCustomTabCompletions(TabCompletionsBuilder tab) {
+        tab.add((Set<String>) MythicMobsBridge.getSkillNames());
```

Comment:
Should remove the `Set<String>` cast here, as these methods take an `Iterable` now